src/ConfigEditor.test.tsx

```diff
@@ -16,169 +16,178 @@ const setUpMockBackendServer = (mockBackendSrv: { put: () => void; post: () => v
+        setUpMockBackendServer({
+          put: jest.fn().mockResolvedValue({ datasource: {} }),
+          post: jest.fn().mockResolvedValue({ externalId: 'fake-external-id' }),
+        });
+    
+        render(
+          <ConfigEditor
+            {...props}
+            options={{
+              ...props.options,
+              jsonData: {
+                ...props.options.jsonData,
+                authType: AwsAuthType.GrafanaAssumeRole,
+              },
+            }}
+          />
+        );
+    
+        expect(screen.queryByText('Grafana Assume Role')).toBeInTheDocument();
+        const instructionsButton = await screen.findByRole('button', {
+          name: /How to create an IAM role for grafana to assume/i,
+        });
+        await userEvent.click(instructionsButton);
+        expect(screen.queryByText('fake-external-id')).toBeInTheDocument();
+      });
+    
+      it('gracefully handles when the fetch for external id throws an error', async () => {
+        setUpMockBackendServer({
+          put: jest.fn().mockResolvedValue({ datasource: {} }),
+          post: jest.fn().mockRejectedValue('the server exploded for some reason'),
+        });
+    
+        render(
+          <ConfigEditor
+            {...props}
+            options={{
+              ...props.options,
+              jsonData: {
+                ...props.options.jsonData,
+                authType: AwsAuthType.GrafanaAssumeRole,
+              },
+            }}
+          />
+        );
+    
+        expect(screen.queryByText('Grafana Assume Role')).toBeInTheDocument();
+        const instructionsButton = await screen.findByRole('button', {
+          name: /How to create an IAM role for grafana to assume/i,
+        });
+        await userEvent.click(instructionsButton);
+        expect(screen.queryByText('External Id is currently unavailable')).toBeInTheDocument();
+      });
+    
+      it('gracefully handles when the fetch for external id return an empty string', async () => {
+        setUpMockBackendServer({
+          put: jest.fn().mockResolvedValue({ datasource: {} }),
+          post: jest.fn().mockResolvedValue({ externalId: '' }),
+        });
+    
+        render(
+          <ConfigEditor
+            {...props}
+            options={{
+              ...props.options,
+              jsonData: {
+                ...props.options.jsonData,
+                authType: AwsAuthType.GrafanaAssumeRole,
+              },
+            }}
+          />
+        );
+    
+        expect(screen.queryByText('Grafana Assume Role')).toBeInTheDocument();
+        const instructionsButton = await screen.findByRole('button', {
+          name: /How to create an IAM role for grafana to assume/i,
+        });
+        await userEvent.click(instructionsButton);
+        expect(screen.queryByText('External Id is currently unavailable')).toBeInTheDocument();
+      });
+    }
+    )}
+
+  run('QueryEditorForm with awsDatasourcesNewFormStyling disabled');
```

Comment:
I thought this was a cleaner solution than copying the tests into another file or below the existing ones for old form. This will rerun the same tests just with the different feature flag. One drawback of this is that you can't run only specific tests, but I don't think this will be a problem  as I don't expect us to make changes to these files a lot before we remove the feature flag